src/traffic_server/traffic_server.cc

```diff
@@ -2215,6 +2215,21 @@ main(int /* argc ATS_UNUSED */, const char **argv)
       REC_ReadConfigInteger(num_of_udp_threads, "proxy.config.udp.threads");
     }
 
+    ats_scoped_str server_ports(255);
+    *server_ports = '\0';
+    REC_ReadConfigString(server_ports, "proxy.config.http.server_ports", 255);
+    bool quic_server_port = strstr(server_ports, "/quic/") != server_ports;
+
+#if TS_USE_QUIC == 0
+    if (quic_server_port) {
+      Fatal("proxy.config.http.server_ports listening for quic but traffic_server wasn't built with quic support.");
```

Comment:
This check could be done as a part of this block, if we want to have a specialized message for QUIC (the block already has code for unrecognized options).
https://github.com/apache/trafficserver/blob/fc5918e5a8bb1eb88b98689c463c3574a64e8bef/src/records/RecHttp.cc#L367-L463

Reply:
@maskit Yeah good suggestion! I tried looking into the records code to see a good place to put this logic in but couldn't find a good place. I see if I can make progress with your suggestion here.